src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -1289,6 +1275,40 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
         .defaultIfEmpty(Optional.empty());
   }
 
+
+  private Mono<TransportCall> fetchTransportCallById(String transportCallId) {
+    if (transportCallId == null) return Mono.empty();
+    return transportCallRepository.findById(transportCallId);
+  }
+
+  private Mono<Optional<Map<String, String>>> fetchImportExportVoyageNumberByTransportCallId(
+      TransportCall transportCall) {
+    if (transportCall == null) return Mono.just(Optional.empty());
+
+    return voyageRepository
+        .findById(transportCall.getImportVoyageID())
```

Comment:
I think it would be a good idea to have a null check for `transportCall.getImportVoyageID` before querying the `voyageRepository`?

Reply:
added check